client/engine/messageservice/p2p-message-service/service.go

```diff
@@ -179,3 +173,35 @@ func (s *P2PMessageService) Close() error {
 	s.p2pHost.RemoveStreamHandler(PROTOCOL_ID)
 	return s.p2pHost.Close()
 }
+
+// generateKey generates a ECDSA key deterministically from the given address
+// TOODO: This is probably a security hole, but it's good enough for now
+func generateKey(address common.Address) p2pcrypto.PrivKey {
+
+	messageKey, _, err := p2pcrypto.GenerateECDSAKeyPair(createReader(address))
+	if err != nil {
+		panic(err)
```

Comment:
Should we bubble up these errors?

Reply:
Yup! In general I need to do a pass on this PR to clean up error handling. I've just been using `panic`s so I don't miss errors while building/debugging.